jaworski_cellseg/label_counting_widget.py

```diff
@@ -18,23 +59,43 @@ def create_label_counting_widget(viewer: napari.Viewer):
             "label": "Total count",
             "enabled": False,
         },  # Make the textbox read-only
+        save_results={
+            "widget_type": "FileEdit",
+            "mode": "w",
+            "filter": "*.csv",
+        },  # Dynamic choices will be set later
     )
     def cell_counting_widget(
-        label_layer_name: str = None,
-        binary_layer_name: str = None,
-        total_count: str = "",
+        label_layer_name: Optional[str] = None,
+        binary_layer_name: Optional[str] = None,
+        total_count: Optional[str] = "",
+        save_results: Optional[str] = None,
     ):
-        print(f"Selected label layer: {label_layer_name}")
-        print(f"Total count: {total_count}")
         labeled_layer = viewer.layers[label_layer_name]
         binary_mask = viewer.layers[binary_layer_name]
+
         ## filter lables outside the binary mask
         result_image = labeled_layer.data * binary_mask.data
         filtered_labels_regions = regionprops(result_image)
         count = len(filtered_labels_regions)
         cell_counting_widget.total_count.value = str(count)
+
         viewer.add_labels(result_image, name=f"{count} Filtered Labled layer")
 
+        file_path = cell_counting_widget.save_results.value
+        if file_path:
+            path = Path(file_path)
+            # Ensure the file has a ".csv" extension
+            if not path.suffix:
+                path = path.with_suffix(
+                    ".csv"
+                )  # Add default ".csv" extension if missing
+
+            with path.open("w") as f:
+                generate_statistics(filtered_labels_regions, physical_sizes, path)
```

Comment:
Because `generate_statistics` calls `df.to_csv(path)` the context manager is not needed.
```suggestion
            generate_statistics(filtered_labels_regions, physical_sizes, path) 
```